ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -107,6 +108,7 @@ public void run() {
     try {
       do {
         long startedAt = System.currentTimeMillis();
+        boolean err = false;
         launchedJob = true;
```

Comment:
that doesn't make sense ( https://github.com/apache/hive/commit/6b0139188aba6a95808c8d1bec63a651ec9e4bdc) why is it set to true even before the submit??

Reply:
I can set the default value to false for this.

Reply:
also `launchedJob` is treated as a successfully executed job, do we need an extra err variable in this case?

Reply:
To implement backoff we do need err to differentiate weather its an error or we dont have any jobs to launch

Reply:
@deniskuzZ That is because all the catch blocks were setting launchedJob to true, or in other words, only the executor.submit() result could set it to false. By initializing it with true there was no need to set it in every catch block.

Reply:
@veghlaci05 I've changed the initial value to false now, I think it shouldnt break any flow as we have err flag to differentiate the exception cases. Do you see any issues with setting it to false?

Reply:
@akshat0395 No, with the introduction of the separate err variable I think setting it to false by default is correct.

Reply:
shouldn't we sleep without backoff if no work is there?

Reply:
Thread does sleep if there is no work, if you see the if block `((!launchedJob || err) && !stop.get())` when launchedJob is false we send the thread to sleep for a default time for no backoff

Reply:
yes, sorry, wanted to delete this question